### PR TITLE
feat(bridge): Replace memory store with MongoDB (#6688)

### LINKED ISSUE #6688: Bridge: Make use of MongoDB for OIDC data
## User Story

As a user, I'd like to have multiple replicas of the Bridge and use an SSO integration. 

## Technical details

If a user does the first login with one replica of the Bridge, they need to re-authenticate if their follow-up session will hit a different replica. To solve this problem, the Bridge should store OAuth data in MongoDB inside a dedicated database.
The whole OAuth session that will be stored is made by:
- access token
- refresh token
- state
- code verifier
- nonce
- expire_in
- id_token

To avoid continuous reads from the database, a cache should be used with a fixed size to avoid OOM errors.

## Acceptance Criteria
- [x] The behavior is enabled iff the `oauth` feature flag is enabled
- [x] `bridge-bff` will create a new database in mongo to store this data (if it doesn't exist yet)
- [ ] A cache system is used to prevent continuous reads from the database
- [ ] The cache should be an LRU which size can be configured via helm value

## DoD
- unit tests

### PR DESCRIPTION
**Pull request by Kirdock**
Closes #6688 
Introduces three environment variables:
- `SESSION_VALIDATING_TIMEOUT_MIN`: Time when a document inside the `validation` collection is deleted. Default is 60.
- `OAUTH_SESSION_SECRET`: Secret to hash the user session. The secret is generated during installation.
- `OAUTH_DATABASE_ENCRYPT_SECRET`: Secret to encrypt the data stored in the MongoDB (session and validation data like nonce and code verifier). This secret must have a length of 32. The secret is generated during installation.

Documentation about changing the new secrets: https://github.com/keptn/keptn.github.io/pull/1025

How to test locally:
- Have a MongoDB in place. `docker run --env MONGODB_DATABASE=keptn --env MONGODB_ROOT_USERNAME=root --env MONGODB_ROOT_PASSWORD=keptnasdf --name mydatabase -p 27017:27017 -d bitnami/mongodb` and create a user that can only access the keptn database.
- Run the container.
- Setup SSO for bridge server https://github.com/keptn/keptn/tree/master/bridge#setting-up-login-via-openid
- Adjust the environment variables for the bridge server. `MONGODB_USER`, `MONGODB_PASSWORD`, `MONGODB_HOST=localhost:27017`, `MONGODB_DATABASE`
- in `keptn/bridge/config/oauth` add the secrets `session_secret` and `database_encrypt_secret` (this one needs to be 32 characters long)

How to test in cluster: 
- Run the installer or just deploy the bridge to the cluster (needs to be done to set/create the right environment variables and generation of secrets)
- Setup SSO for bridge server https://github.com/keptn/keptn/tree/master/bridge#setting-up-login-via-openid

Signed-off-by: Klaus Strießnig <k.striessnig@gmail.com>

### CODE REVIEW

File: installer/manifests/keptn/charts/control-plane/templates/core.yaml [hunk 1]
```diff
@@ -264,9 +288,26 @@ spec:
...
             - name: SESSION_TIMEOUT_MIN
               value: "{{ .Values.bridge.oauth.sessionTimeoutMin}}"
+            # Timeout of validation data (state, nonce, code verifier) during login
+            - name: SESSION_VALIDATING_TIMEOUT_MIN
+              value: "{{ .Values.bridge.oauth.sessionValidatingTimeoutMin}}"
             # Correlates to trust proxy number of hops as defined at http://expressjs.com/en/guide/behind-proxies.html
             - name: TRUST_PROXY
               value: "{{ .Values.bridge.oauth.trustProxy }}"
+            - name: MONGODB_HOST
+              value: keptn-mongo:27017
```
Comment (thisthat) [inline installer/manifests/keptn/charts/control-plane/templates/core.yaml:298]: We allow to specify an external mongo installation, please use the following like we have for mongodatastore
```suggestion
              value: '{{ .Release.Name }}-{{ .Values.mongo.service.nameOverride }}:{{ .Values.mongo.service.port }}'
```

File: installer/manifests/keptn/charts/control-plane/templates/core.yaml [hunk 2]
```diff
@@ -264,9 +288,26 @@ spec:
...
+                secretKeyRef:
+                  name: mongodb-credentials
+                  key: mongodb-user
+            - name: MONGODB_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: mongodb-credentials
+                  key: mongodb-password
+            - name: MONGODB_DATABASE
+              value: openid
```
Comment (thisthat) [inline installer/manifests/keptn/charts/control-plane/templates/core.yaml:310]: I would make this configurable with an optional. Please, also add this inside the `value.yaml` file around line 9.
```suggestion
              value: {{ .Values.mongo.auth.bridgeAuthDatabase | default "openid" }}
```

File: installer/manifests/keptn/charts/control-plane/templates/core.yaml [hunk 3]
```diff
@@ -7,9 +7,16 @@
 {{- $bridgePassword := (randAlphaNum 20) | b64enc | quote }}
 {{- $bridgesecret := (lookup "v1" "Secret" .Release.Namespace "bridge-credentials") }}
 {{- if $bridgesecret }}
-{{- $bridgePassword = index $bridgesecret.data "BASIC_AUTH_PASSWORD" }}
+  {{- $bridgePassword = index $bridgesecret.data "BASIC_AUTH_PASSWORD" }}
```
[the remark this answers is not in the document]
Comment (thisthat) [reply installer/manifests/keptn/charts/control-plane/templates/core.yaml:10]: I miss the if, please ignore my comment

**Review by mowies — APPROVED** [review]
LGTM!

**Review by laneli — APPROVED** [review]

**Pull request merged**